Feature/learn html

reijoshua opened the pull request:
2.1 Release WIP

- Breaks up server into routers.
    - Allows for more specific control of middlewares and access control to static resources
- Adds benchmark tests for all major pages and routers
- Adds Deno general tasks 
    - deno task format
    - deno task start
    - deno task compile
- Add global element functions to the global scope on the window.
- General comment for better documentation when I merge this into the fork template.

Commit:
WIP

Basic grid mostly done, fetch content not working still playing around with how I want to approach that. Sub routes with a appended /value can't reference resource includes with nested resources that call public assets in the more general context. So when rendering the nested footer within the content page is not rendering the icons in the footer. content/assets/...

Commit:
Dynamic Grid

Click triggers fetch which returns Json with html. Event is dispatched with data from the server which is picked up by the event handler. Dynamically loaded content Js is init events etc. in this case the content close button to reveal the gird and clear the content.

Commit:
Merge branch 'master' into feature/Pre_release_product_grid

Commit:
Merge branch 'develop' into feature/Pre_release_product_grid

Commit:
Content Grid WIP #2

Content Gird
Fetch the article by Id, return article data, and populate the content section.
Adds Mutation Observer to watch for the content update then enact any needed Js on the newly added elements. This handles the issues with deferred event handling as we use our mutation events to enact our new element functionality. For example the close button for the article to re-show the content grid and currently clear the content section.

reijoshua committed:
Routes to Controllers

Broke down routes in jsLand.js to the controllers folder as well as created the controllers bundle for managing controller exports. Add new controller calls to the main js file. All we pass is the server object to set up the routes.

Commit:
Formatting

Formatting updates and some basic tasks.

reijoshua committed:
Wip

Commit:
WIP Break up server into routers

Routers, grid inline caching of content if present load from inline-cache no fetch needed. Wrote tests for both routers.

reijoshua committed:
WIP

Require comment for later use, another variant on the masthead.

reijoshua committed:
WIP

Learn html among some other refactors adds global init for some elements needed in all templates head footer.

reijoshua committed:
Adds comments

Adds JSDOC comments for all major methods and functionality. Also adds per module benchmark tests for many of the site main routers and pages.

reijoshua committed:
Add JSON/Html

You can hit any of the content endpoints with either a html path or json path to be served the two different formats of a given content page. We use the json version to remote server on a content grid. That fetch is then inline cached on the card so we don't need to make any additional requests for that content.

RedVanJosh commented:
You can hit any of the content endpoints with either a html path or json path to be served the two different formats of a given content page. We use the json version to remote server on a content grid. That fetch is then inline cached on the card so we don't need to make any additional requests for that content.

reijoshua committed:
Ran the format task

Updating format across the code base. Some new images and a few tweaks. Still an outstanding issue with static content in the nested routes.

reijoshua committed:
Some clean up and refactoring

Also adds tools page with empty content for now and refactors the content route to be content/grid

Commit:
Add some additional styling on the LearnHtml page.

Add styling some formatting updates. html/json view of content pages is working as expected.

reijoshua committed:
Prototype run CLI Command

Runs command in sub process and returns result.

reijoshua committed:
add tools.css

reijoshua committed:
WIP Flash Cards

reijoshua committed:
WIP Flash Cards

reijoshua committed:
Merge remote-tracking branch 'origin/feature/LearnHtml' into feature/LearnHtml
